python/tests/test_random_forest.py

```diff
@@ -419,6 +422,10 @@ def test_random_forest_classifier_spark_compat(
             rf.setLeafCol("leafId")
             assert rf.getLeafCol() == "leafId"
         else:
+            assert isinstance(rf, RandomForestClassifier)
+            # reduce the number of GPUs for toy dataset to avoid empty partition
+            gpu_number = 1
+            rf.num_workers = gpu_number
```

Comment:
Should probably just use `gpu_number = min(gpu_number, 1)` to be consistent with all the other changes.

Reply:
Same here.

Reply:
Revised to gpu_number = min(gpu_number, 2), and increased the size of the toy dataset.